presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -626,15 +653,49 @@ void PrestoServer::yieldTasks() {
   }
 }
 
+#ifdef __linux__
+class BatchThreadFactory : public folly::NamedThreadFactory {
+ public:
+  explicit BatchThreadFactory(const std::string& name)
+      : NamedThreadFactory{name} {}
+
+  std::thread newThread(folly::Func&& func) override {
+    return folly::NamedThreadFactory::newThread(
+        [func_2 = std::move(func)]() mutable {
+          sched_param param;
+          param.sched_priority = 0;
+          int error =
+              pthread_setschedparam(pthread_self(), SCHED_BATCH, &param);
+          VELOX_CHECK(
+              error == 0,
+              "Failed to set a thread priority: {}",
+              std::to_string(error));
+          func_2();
+        });
+  }
+};
+#endif
+
 void PrestoServer::initializeThreadPools() {
   const auto hwConcurrency = std::thread::hardware_concurrency();
   auto* systemConfig = SystemConfig::instance();
 
   const auto numDriverCpuThreads = std::max<size_t>(
       systemConfig->driverNumCpuThreadsHwMultiplier() * hwConcurrency, 1);
+
+  std::shared_ptr<folly::NamedThreadFactory> threadFactory;
+  if (systemConfig->driverThreadsBatchSchedulingEnabled()) {
+#ifdef __linux__
+    threadFactory = std::make_shared<BatchThreadFactory>("Driver");
+#else
```

Comment:
Can we log a warning instead of fail the server start if the non-linux platform doesn't this?

Reply:
The property is `false` by default and shouldn't be set on systems other than Linux. Failing loud when somebody is trying to set it on systems other than Linux seems to less error prone.